pkg/apis/metering/v1/metering_config.go

```diff
@@ -0,0 +1,624 @@
+	TLS                 *TLSSpec                 `json:"tls,omitempty"`
+	Monitoring          *MonitoringSpec          `json:"monitoring,omitempty"`
+	Permissions         *Permissions             `json:"permissions,omitempty"`
+	OpenshiftReporting  *OpenshiftReporting      `json:"openshift-reporting,omitempty"`
+	Ghostunnel          *Ghostunnel              `json:"__ghostunnel,omitempty"`
+	ReportingOperator   *ReportingOperator       `json:"reporting-operator,omitempty"`
+	Presto              *Presto                  `json:"presto,omitempty"`
+	Hive                *Hive                    `json:"hive,omitempty"`
+	Hadoop              *Hadoop                  `json:"hadoop,omitempty"`
+
+	DisableOCPFeatures bool `json:"disableOCPFeatures,omitempty"`
+	LogHelmTemplate    bool `json:"logHelmTemplate,omitempty"`
+}
+
+/*
+Start of structures re-used throughout the top-level keys
+*/
+
+type Labels struct {
+}
+type Image struct {
+	PullPolicy string `json:"pullPolicy,omitempty"`
+	Repository string `json:"repository,omitempty"`
+	Tag        string `json:"tag,omitempty"`
+}
+type GeneralTLS struct {
+	Certificate   string `json:"certificate,omitempty"`
+	Key           string `json:"key,omitempty"`
+	CaCertificate string `json:"caCertificate,omitempty"`
+	CreateSecret  bool   `json:"createSecret"`
+	Enabled       bool   `json:"enabled"`
+	SecretName    string `json:"secretName,omitempty"`
+}
+type GeneralJVM struct {
+	G1HeapRegionSize               int    `json:"G1HeapRegionSize,omitempty"`
+	ConcGCThreads                  int    `json:"concGCThreads,omitempty"`
+	InitiatingHeapOccupancyPercent int    `json:"initiatingHeapOccupancyPercent,omitempty"`
+	MaxGcPauseMillis               int    `json:"maxGcPauseMillis,omitempty"`
+	ParallelGCThreads              int    `json:"parallelGCThreads,omitempty"`
+	InitialRAMPercentage           int    `json:"initialRAMPercentage,omitempty"`
+	MaxRAMPercentage               int    `json:"maxRAMPercentage,omitempty"`
+	MinRAMPercentage               int    `json:"minRAMPercentage,omitempty"`
+	PermSize                       string `json:"permSize,omitempty"`
+	ReservedCodeCacheSize          string `json:"reservedCodeCacheSize,omitempty"`
+	MaxCachedBufferSize            int    `json:"maxCachedBufferSize,omitempty"`
+	MaxDirectMemorySize            int    `json:"maxDirectMemorySize,omitempty"`
+}
+
+/*
+End of structures re-used throughout the top-level keys
+*/
+
+type ReportingOperator struct {
+	Spec ReportingOperatorSpec `json:"spec,omitempty"`
+}
+type ReportingOperatorSpec struct {
+	Affinity       *corev1.Affinity                  `json:"affinity,omitempty"`
+	Annotations    map[string]string                 `json:"annotations,omitempty"`
+	Config         *ReportingOperatorConfig          `json:"config,omitempty"`
+	Labels         *Labels                           `json:"labels,omitempty"`
+	NodeSelector   *corev1.NodeSelector              `json:"nodeSelector,omitempty"`
```

Comment:
@chancez we need to change this type to something other than `corev1.NodeSelector` or update the charts/manifests that handle the `nodeSelector` key as a list of key/value pairs.

Reply:
This is what is produced with the `corev1.NodeSelector` type:
```yaml
apiVersion: metering.openshift.io/v1
kind: MeteringConfig
...
  presto:
    spec:
      coordinator:
        nodeSelector:
          nodeSelectorTerms: null
      worker:
        nodeSelector:
          nodeSelectorTerms: null
...
```